src/management-system-v2/lib/data/legacy/iam/roles.js

```diff
@@ -2,37 +2,100 @@ import { v4 } from 'uuid';
 import store from '../store.js';
 import { roleMigrations } from './migrations/role-migrations.js';
 import { mergeIntoObject } from '../../../helpers/javascriptHelpers';
-import { roleMappingsMetaObjects } from './role-mappings.js';
+import { addRoleMappings, roleMappingsMetaObjects } from './role-mappings.js';
+import { ApiData, ApiRequestBody } from '@/lib/fetch-data';
+import Ability, { UnauthorizedError } from '@/lib/ability/abilityHelper';
+import { toCaslResource } from '@/lib/ability/caslAbility';
+import { adminRules } from '@/lib/authorization/caslRules.ts';
+import { developmentRoleMappingsMigrations } from './migrations/role-mappings-migrations.js';
+
+let firstInit = !global.roleMetaObjects;
 
 /** @type {any} - object containing all roles */
-export let roleMetaObjects = {};
+export let roleMetaObjects = global.roleMetaObjects || (global.roleMetaObjects = {});
+
+/**
+ * initializes the roles meta information objects
+ */
+export function init() {
+  if (!firstInit) return;
+
+  // get roles that were persistently stored
+  const storedRoles = store.get('roles');
+
+  // set roles store
+  store.set('roles', 'roles', storedRoles);
+
+  // migrate roles
+  roleMigrations.forEach((role) => {
+    const index = storedRoles.findIndex((storedRole) => storedRole.name === role.name);
+    if (index >= 0) return;
+
+    const { id: roleId, name } = addRole(role, new Ability(adminRules()));
+
+    if (process.env.NODE_ENV === 'development') {
+      const roleMappings = developmentRoleMappingsMigrations
+        .filter((mapping) => mapping.roleName === name)
+        .map((mapping) => ({
+          roleId,
+          userId: mapping.userId,
+        }));
+
+      addRoleMappings(roleMappings, new Ability(adminRules()));
```

Comment:
Is it unproblematic that these are async functions?

Reply:
I forgot to remove async form addRole. I fixed it in one of my last commits